lib/resource-providers/rds.ts

```diff
@@ -0,0 +1,55 @@
+import { CfnOutput } from "aws-cdk-lib";
+import * as rds from "aws-cdk-lib/aws-rds";
+import { GlobalResources, ResourceContext, ResourceProvider } from "../spi";
+import {IClusterEngine, IDatabaseCluster, InstanceProps} from "aws-cdk-lib/aws-rds";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import {IVpc} from "aws-cdk-lib/aws-ec2";
+
+
+export interface RdsInstanceProps {
+  readonly name?: string;
+  readonly rdsProps?: Omit<rds.DatabaseInstanceProps, "storageEncryptionKey">;
+  readonly rdsEncryptionKeyResourceName?: string;
+}
+
+export interface AuroraClusterProps {
+  readonly name?: string;
+  readonly clusterProps?: Omit<rds.DatabaseClusterProps, "engine" | "instanceProps">
+  readonly instanceProps?: InstanceProps;
+  readonly auroraEngine: IClusterEngine;
+}
+
+export class AuroraClusterProvider
```

Comment:
I am unclear on credentials for this instance. How is the connection made? Especially if it is credentials driven, are those in param store?

Reply:
Still in progress.

Debating on weather to return credentials as part of the output OR just implicitly use it in the backstage addon

Reply:
Credentials for the instance are stored in Secrets Store, and are retrievable through `auroraInstance.secret` after creation.